src/main/java/com/epam/lowcost/DAO/FlightDAOImpl.java

```diff
@@ -0,0 +1,151 @@
+package com.epam.lowcost.DAO;
+
+import com.epam.lowcost.model.Flight;
+import com.epam.lowcost.model.Plane;
+import com.epam.lowcost.util.DateFormatter;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FlightDAOImpl implements FlightDAO {
+    private DataSource dataSource;
+
+    public FlightDAOImpl(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+
+    @Override
+    public List<Flight> getAllFlights() {
+        List<Flight> flights = new ArrayList<>();
+        try (Connection conn = dataSource.getConnection();
+             Statement stmt = conn.createStatement();
+             ResultSet rs = stmt.executeQuery("SELECT * FROM FLIGHTS JOIN  PLANES " +
+                     "ON FLIGHTS.plane_id = PLANES.id WHERE  FLIGHTS.ISDELETED = false ")) {
+            while (rs.next()) {
+                flights.add(extractFlightFromRS(rs));
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return flights;
+    }
+
+    @Override
+    public Flight getById(Long id) {
+        Flight flight = new Flight();
```

Comment:
you can return null and don't create new entity

Reply:
fixed in last commit